Backend/app/rider-platform/rider-app/Main/src/Storage/Queries/Person/PersonDefaultEmergencyNumber.hs

```diff
@@ -43,12 +44,31 @@ findAllByPersonId (Id personId) = findAllWithKV [Se.Is BeamPDEN.personId $ Se.Eq
 findAllByContactPersonId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => Id Person -> m [PersonDefaultEmergencyNumber]
 findAllByContactPersonId (Id contactPersonId) = findAllWithKV [Se.Is BeamPDEN.contactPersonId $ Se.Eq (Just contactPersonId)]
 
-updateEmergencyContactPersonId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => DbHash -> Id Person -> m ()
-updateEmergencyContactPersonId dbHash (Id personId) = do
+updateEmergencyContactPersonId :: (MonadFlow m, CacheFlow m r, EsqDBFlow m r) => DbHash -> Id Person -> Id DMerchant.Merchant -> m ()
+updateEmergencyContactPersonId dbHash (Id personId) (Id merchantId) = do
   updateWithKV
     [ Se.Set BeamPDEN.contactPersonId (Just personId)
     ]
-    [ Se.Is BeamPDEN.mobileNumberHash $ Se.Eq dbHash
+    [ Se.Is BeamPDEN.mobileNumberHash $ Se.Eq dbHash,
+      Se.Is BeamPDEN.merchantId $ Se.Eq (Just merchantId)
```

Comment:
What about older contacts for them merchantId would be null.

Reply:
Yes, we will backfill based on personId id for that.